sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -119,11 +120,13 @@ class AstBuilder extends DataTypeAstBuilder
   override def visitCompoundOrSingleStatement(
       ctx: CompoundOrSingleStatementContext): CompoundBody = withOrigin(ctx) {
     Option(ctx.singleCompoundStatement()).map { s =>
+      if (!SQLConf.get.sqlScriptingEnabled) {
```

Comment:
Why don't you get the current config value from the active config obj `conf`? See other examples in the same file like `conf.getConf(SQLConf.ENABLE_DEFAULT_COLUMNS)`

Reply:
This will be rebased on: https://github.com/apache/spark/pull/47609. Can you please leave comments there first, and then we can get back to this one.